src/rules/consistent-type-specifier-style.js

```diff
@@ -0,0 +1,227 @@
+      url: docsUrl('consistent-type-specifier-style'),
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'string',
+        enum: ['prefer-inline', 'prefer-top-level'],
+        default: 'prefer-inline',
+      },
+    ],
+  },
+
+  create(context) {
+    const sourceCode = context.getSourceCode();
+
+    if (context.options[0] === 'prefer-inline') {
+      return {
+        ImportDeclaration(node) {
+          if (node.importKind === 'value' || node.importKind == null) {
+            return;
+          }
+
+          if (
+            // no specifiers (import type {} from '') have no specifiers to mark as inline
+            node.specifiers.length === 0 ||
+            (node.specifiers.length === 1 &&
+              // default imports are both "inline" and "top-level"
+              (node.specifiers[0].type === 'ImportDefaultSpecifier' ||
+                // namespace imports are both "inline" and "top-level"
+                node.specifiers[0].type === 'ImportNamespaceSpecifier'))
+          ) {
+            return;
+          }
+
+          context.report({
+            node,
+            message: 'Prefer using inline {{kind}} specifiers instead of a top-level {{kind}}-only import.',
+            data: {
+              kind: node.importKind,
+            },
+            fix(fixer) {
+              const fixes = [];
+
+              const kindToken = sourceCode.getFirstToken(node, { skip: 1 });
+              if (kindToken) {
+                fixes.push(fixer.remove(kindToken));
+              }
+
+              for (const specifier of node.specifiers) {
+                fixes.push(fixer.insertTextBefore(specifier, `${node.importKind} `));
+              }
+
+              return fixes;
+            },
+          });
+        },
+      };
+    }
+
+    // prefer-top-level
+    return {
+      ImportDeclaration(node) {
+        if (
+          // already top-level is valid
+          node.importKind === 'type' ||
+          node.importKind === 'typeof' ||
+          // no specifiers (import {} from '') cannot have inline - so is valid
+          node.specifiers.length === 0 ||
+          (node.specifiers.length === 1 &&
+            // default imports are both "inline" and "top-level"
+            (node.specifiers[0].type === 'ImportDefaultSpecifier' ||
+              // namespace imports are both "inline" and "top-level"
+              node.specifiers[0].type === 'ImportNamespaceSpecifier'))
+        ) {
+          return;
+        }
+
+        const typeSpecifiers = [];
+        const typeofSpecifiers = [];
+        const valueSpecifiers = [];
+        let defaultSpecifier = null;
+        for (const specifier of node.specifiers) {
+          if (specifier.type === 'ImportDefaultSpecifier') {
+            defaultSpecifier = specifier;
+            continue;
+          } else if (specifier.type !== 'ImportSpecifier') {
+            continue;
```

Comment:
also this one?

Reply:
We could actually delete this check entirely.
It's covered above by the checks on line 80-84.
It's a syntax error to have a namespace specifier with any other kind of specifier.

This is just me doing my due diligence because I'm used to covering all of the bases due to TS types.

Would you like me to delete it?

Reply:
if it's not used, then it's probably best to delete it.